auth/gcloud/aio/auth/session.py

```diff
@@ -114,6 +114,7 @@ def session(self) -> aiohttp.ClientSession:
                 timeout = aiohttp.ClientTimeout(self._timeout)
                 self._session = aiohttp.ClientSession(connector=connector,
                                                       timeout=timeout)
+            assert self._session is not None
```

Comment:
What's the motivation of these two checks?

Reply:
>It also modifies the typing for BaseSession to ensure that BaseSession.session returns a Session object for all subclasses, as a None value should be made impossible.

That being said, the typing changes I made to `gcloud-aio-auth` are not actually necessary. Unfortunately this is a case of `pre-commit` running quite differently on my machine than in CI as there are far more false positives of typing errors on my machine. Over the course of my changes I still held the believe that `mypy` had to be satisfied here, `mypy` being unsure that the object is not `Optional` as the parent class suggests and necessitating an explicit non-`None` check that `mypy` respects.

Reply:
(I have removed these typing changes to `gcloud-aio-auth`)

Reply:
Ah, I hadn't linked that comment to the `asserts`, that makes sense (TIL about [that](https://github.com/python/typing/issues/645), my guess for how to accomplish this would have been to [use `typing.cast` somehow](https://docs.python.org/3/library/typing.html#typing.cast)).

Definitely curious about the differences in usage between your machine and CI. I feel like we've been getting more reports of that sort of behaviour lately, which is troubling..